frontend/src/components/MapView/LeftPanel/index.tsx

```diff
@@ -124,31 +135,45 @@ const LeftPanel = memo(() => {
 
   // fetch csv data when loading AA page
   React.useEffect(() => {
-    if (tabValue !== Panel.AnticipatoryAction) {
+    if (!isAnticipatoryActionLayer(tabValue)) {
       return;
     }
     dispatch(loadAAData());
   }, [dispatch, tabValue]);
 
   // Add layers to url
   React.useEffect(() => {
-    if (tabValue !== Panel.AnticipatoryAction) {
+    if (!isAnticipatoryActionLayer(tabValue)) {
       return;
     }
 
-    const layer = LayerDefinitions[AALayerId];
+    const selectedLayerId = AALayerIds.find(x => x === tabValue);
+
+    if (!selectedLayerId) {
+      return;
+    }
 
+    const layer = LayerDefinitions[selectedLayerId];
+    console.log('layer', layer);
+    console.log('AALayerInUrl', AALayerInUrl);
     // Add to url when getting to AA tab
-    if (AALayerInUrl !== undefined || !layer) {
+    if (!layer) {
       return;
     }
 
+    // Change AA layer in url
```

Comment:
TODO: Should rework AA layers switch in url, currently not working correctly

Reply:
switch between AA in url is now working but date query no longer appear in url after switch

Reply:
resolved